sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationIdentifierSerializer.java

```diff
@@ -0,0 +1,75 @@
+
+        if (kind == CommunicationIdentifierKind.COMMUNICATION_USER) {
+            Objects.requireNonNull(id);
+            return new CommunicationUserIdentifier(id);
+        }
+
+        if (kind == CommunicationIdentifierKind.CALLING_APPLICATION) {
+            Objects.requireNonNull(id);
+            return new CallingApplicationIdentifier(id);
+        }
+
+        if (kind == CommunicationIdentifierKind.PHONE_NUMBER) {
+            Objects.requireNonNull(identifier.getPhoneNumber());
+            Objects.requireNonNull(identifier.getId());
+            return new PhoneNumberIdentifier(identifier.getPhoneNumber()).setId(identifier.getId());
+        }
+
+        if (kind == CommunicationIdentifierKind.MICROSOFT_TEAMS_USER) {
+            Objects.requireNonNull(identifier.getMicrosoftTeamsUserId());
+            Objects.requireNonNull(identifier.getCloudEnvironmentModel());
+            return new MicrosoftTeamsUserIdentifier(identifier.getMicrosoftTeamsUserId(), identifier.isAnonymous())
+                .setId(identifier.getId())
+                .setCloudEnvironment(CommunicationCloudEnvironment.fromModel(identifier.getCloudEnvironmentModel()));
+        }
+
+        Objects.requireNonNull(id);
+        return new UnknownIdentifier(id);
+    }
+
+    public static CommunicationIdentifierModel serialize(CommunicationIdentifier identifier) {
+        if (identifier instanceof CommunicationUserIdentifier) {
+            return new CommunicationIdentifierModel()
+                .setKind(CommunicationIdentifierKind.COMMUNICATION_USER)
+                .setId(((CommunicationUserIdentifier) identifier).getId());
+        }
+
+        if (identifier instanceof CallingApplicationIdentifier) {
+            return new CommunicationIdentifierModel()
+                .setKind(CommunicationIdentifierKind.CALLING_APPLICATION)
+                .setId(((CallingApplicationIdentifier) identifier).getId());
+        }
+
+        if (identifier instanceof PhoneNumberIdentifier) {
+            return new CommunicationIdentifierModel()
+                .setKind(CommunicationIdentifierKind.PHONE_NUMBER)
+                .setPhoneNumber(((PhoneNumberIdentifier) identifier).getPhoneNumber())
+                .setId(identifier.getId());
+        }
+
+        if (identifier instanceof MicrosoftTeamsUserIdentifier) {
+            MicrosoftTeamsUserIdentifier teamsUserIdentifier = (MicrosoftTeamsUserIdentifier) identifier;
+            return new CommunicationIdentifierModel()
+                .setKind(CommunicationIdentifierKind.MICROSOFT_TEAMS_USER)
+                .setMicrosoftTeamsUserId(teamsUserIdentifier.getUserId())
+                .setIsAnonymous(teamsUserIdentifier.isAnonymous())
+                .setId(teamsUserIdentifier.getId())
+                .setCloudEnvironmentModel(new CommunicationCloudEnvironmentModel(teamsUserIdentifier.getCloudEnvironment().toString()));
+        }
+
+        return new CommunicationIdentifierModel()
```

Comment:
Let's follow C# logic and do another if for Unknown, and then throw at the end
https://github.com/Azure/azure-sdk-for-net/blob/735d6d055b22086cc4c457451945174e985735e4/sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs#L73

The reason is that otherwise it's easy to add a new identifier but forgot to update the serializer. When we throw, we catch that much faster.